### PR TITLE
OADP-3051: Always restore replicaset

### PR DESCRIPTION
**Pull request by sseago**
Always restore replicaset since in some cross-cluster restore scenarios a newly-deployed replicaset will get a different name, breaking the restored pod.

Also, remove MTC-specific RS backup plugin, since the only action there was to mark paused replicasets for restore, which is unnecessary now that we're restoring all of them.

### CODE REVIEW

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by mpryc — APPROVED** [review]
Thanks for clarification offline about ownerRefs. Looks fine, would like someone else as well to ack this.

File: velero-plugins/replicaset/restore.go [hunk 1]
```diff
@@ -38,27 +38,6 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
...
 	common.SwapContainerImageRefs(replicaSet.Spec.Template.Spec.InitContainers, backupRegistry, registry, p.Log, input.Restore.Spec.NamespaceMapping)
 
-	ownerRefs, err := common.GetOwnerReferences(input.ItemFromBackup)
-	if err != nil {
-		return nil, err
-	}
-
-	annotations := replicaSet.Annotations
-
-	// Don't restore ReplicaSet if owned by Deployment and the deployment was not paused
```
Comment (amastbau) [inline velero-plugins/replicaset/restore.go:48]: Is the  replicaset restored before the deployment?

Comment (sseago) [reply velero-plugins/replicaset/restore.go:48]: @amastbau Yes -- I just checked the restore priorities list. They're restored right after pods. Deployments come later in the "everything not called out as high priority" group

**Pull request merged**